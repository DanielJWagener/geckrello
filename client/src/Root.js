// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
