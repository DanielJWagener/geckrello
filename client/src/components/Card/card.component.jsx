import React from "react";
import { ItemTypes } from "../Constants";
import { useDrag } from "react-dnd";
import { connect } from "react-redux";
import { moveCard } from "../../redux/cards/cards.actions";

import "./card.styles.scss";

export function Card({ moveCard, cardId, cardTitle }) {
  // Drag source hook
  const [{ isDragging }, drag] = useDrag({
    item: { type: ItemTypes.CARD },
    end: (item, monitor) => {
      const dropResult = monitor.getDropResult();
      if (item && dropResult) {
        moveCard(cardId, dropResult.listId);
      }
    },
    collect: monitor => ({
      isDragging: !!monitor.isDragging()
    })
  });

  const popupHref = `#${cardId}`;

  return (
    <div className="card">
      <a
        href={popupHref}
        ref={drag}
        className="card__preview"
        style={{
          opacity: isDragging ? 0.5 : 1,
          zIndex: isDragging ? 3 : 3,
          cursor: "move"
        }}
      >
        {cardTitle}
      </a>
    </div>
  );
}

export default connect(null, { moveCard })(Card);
