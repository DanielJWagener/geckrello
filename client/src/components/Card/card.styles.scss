@import '../../sass/abstracts/_variables.scss';

.card {
  &:not(:last-child) {
    margin-bottom: 1rem;
  }

  &:first-child {
    margin-top: 0.5rem;
  }

  z-index: 10;

  &__preview {
    display: block;
    position: relative;
    background-color: #fff;
    border-radius: 0.5rem;
    font-size: 2rem;
    padding: 0 1rem;
    line-height: 3.5rem;
    color: $default-font-color;
    text-decoration: none;
    border-bottom: 1px solid $default-font-color;

    &:hover > .card__modal-link {
      display: block;
      cursor: pointer;
    }
  }

  &__modal-link {
    display: none;
    color: #444;
    text-decoration: none;
    position: absolute;
    right: 5px;
    top: 50%;
    transform: translateY(-50%);

    &:active {
      color: #444;
    }

    &:visited {
      color: #444;
    }
  }
}
