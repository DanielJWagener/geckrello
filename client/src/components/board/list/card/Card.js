import React from "react";
import { ItemTypes } from "../../Constants";
import { useDrag } from "react-dnd";
import { connect } from "react-redux";
import { moveCard } from "../../../../actions";

function Card(props) {
  // Drag source hook
  const [{ isDragging }, drag] = useDrag({
    item: { type: ItemTypes.CARD },
    end: (item, monitor) => {
      const dropResult = monitor.getDropResult();
      if (item && dropResult) {
        props.moveCard(props.cardId, dropResult.listId);
      }
    },
    collect: monitor => ({
      isDragging: !!monitor.isDragging()
    })
  });

  const popupHref = `#${props.cardId}`;

  return (
    <div className="card">
      <a
        href={popupHref}
        ref={drag}
        className="card__preview"
        style={{
          opacity: isDragging ? 0.5 : 1,
          zIndex: isDragging ? 3 : 3,
          cursor: "move"
        }}
      >
        {props.cardTitle}
      </a>
    </div>
  );
}

const mapStateToProps = state => {
  return { cards: state.cards };
};

export default connect(
  mapStateToProps,
  { moveCard }
)(Card);
