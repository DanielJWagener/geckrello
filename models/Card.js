const mongoose = require("mongoose");
const { Schema } = mongoose;
const ChecklistItemSchema = require("./ChecklistItem");

const cardSchema = new Schema({
  title: String,
  description: {
    type: String,
    default: ""
  },
  checklist: [ChecklistItemSchema],
  listHome: String,
  boardHome: String,
  archived: {
    type: Boolean,
    default: false
  }
});

const Card = mongoose.model("cards", cardSchema);

module.exports = Card;
