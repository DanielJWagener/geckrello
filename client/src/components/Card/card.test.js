import { shallow } from "enzyme";
import React from "react";
import Card from "./card.component";
import Root from "../../Root";

it("renders Card component", () => {
  expect(
    shallow(
      <Root>
        <Card />
      </Root>
    )
  ).toMatchSnapshot();
});
