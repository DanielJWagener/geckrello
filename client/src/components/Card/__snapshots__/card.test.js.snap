// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component 1`] = `ShallowWrapper {}`;
